benchmarks/cpu_gpu_times_wrapper.py

```diff
@@ -0,0 +1,93 @@
+import warnings
+
+#from pytest_benchmark.fixture import BenchmarkFixture
+from pytest_benchmark.fixture import FixtureAlreadyUsed
+from pytest_benchmark.stats import Metadata
+from cupyx.time import repeat as cp_repeat
+
+
+class GpuWrapper(object):
+    def __init__(self, wrapped_class,iterations=30, rounds=5, warmup_rounds=10):
+        self.__dict__['wrapped_class'] = wrapped_class
+        self.iterations = iterations
+        self.rounds = rounds
+        self.warmup_rounds = warmup_rounds
+
+
+    def __getattr__(self, attr):
+        #orig_attr = self.wrapped_class.__getattribute__(attr)
+        orig_attr = getattr(self.wrapped_class, attr)
+        if callable(orig_attr):
+            def hooked(*args, **kwargs):
+                result = orig_attr(*args, **kwargs)
+                # prevent wrapped_class from becoming unwrapped
+                if result == self.wrapped_class:
+                    return self
+                return result
+            return hooked
+        else:
+            return orig_attr
+
+    def __setattr__(self, attr, value):
+        setattr(self.wrapped_class, attr, value)
```

Comment:
I think we can safely remove `__setattr__`.

Setting `__setattr__` causes `self.x = y` to always call `__setattr__` and setting an attribute on self has to be done via `self.__dict__["x"] = y` (which is painful and easy to forget to do). On top of this, its hard to imagine why someone doing `self.x = y` on the wrapper object would want the attribute to be set on the wrapped *class*.

Reply:
I need things to be set on the wrapped class because it is the one that does all the communications with the pytest benchmark modules. The wrapper class should not have any attributes besides the class being wrapped. Maybe there is an easier way to dinamically add this functionality on the classes ?

I think that the rapids version, which does not have the `__setattr__` may have issues when manually setting attributes , like in [Asier's code](https://github.com/qutip/qutip-tensorflow/blob/f2eba14d9010545deab1d300001375e72b15c456/tests/test_linear_algebra.py#L97), which I would like to reuse in part.

Notice that if we were to register our wrapper to `pytest` , when doing `benchmark.groups = "x" ` we wouldn't be setting the wrapped `groups`  attribute, which then gets pushed to the `BenchmarkSession`.